src/strace.c

```diff
@@ -2633,10 +2636,16 @@ init(int argc, char *argv[])
 				error_opt_arg(c, lopt, optarg);
 			break;
 		case 's':
-			i = string_to_uint(optarg);
-			if (i < 0 || (unsigned int) i > -1U / 4)
+			if (isdigit(*optarg)) {
+				i = string_to_uint(optarg);
+				if (i < 0 || (unsigned int) i > -1U / 4)
+					error_opt_arg(c, lopt, optarg);
+				max_strlen = i;
+			} else if (strcmp(optarg, "inf") == 0) {
+				truncate_output = false;
+				sysinfo(&info);
```

Comment:
In this branch `max_strlen` is not assigned so it remains initialized with the default value.
This wouldn't be an issue if `max_strlen` wasn't used in several places besides being an argument in `truncation_needed` invocations.

By the way, what will happen if `sysinfo` failed?